### PR TITLE
graph_fetch: reuse clients when contacting graph-builder

### PR DESCRIPTION
**Pull request by vrutkovs**
policy-engine is creating a new http client every time graph is being fetched.
This causes HTP 408 or even 500 errors on a large number of requests
 (~100 rps in my testing). Instead http client should be reused to
 benefit from reqwest connection pooling.

This makes load-testing look much better: 0 errors,
```
Bucket           #      %       Histogram
[0s,     50ms]   44915  99.81%  ##########################################################################
[50ms,   100ms]  82     0.18%   
[100ms,  500ms]  3      0.01%   
[500ms,  1s]     0      0.00%   
[1s,     5s]     0      0.00%   
[5s,     10s]    0      0.00%   
[10s,    +Inf]   0      0.00%  
```

With this PR 100 rps is easily handled and PE starts throwing HTTP 500 and 408 on ~5000 rps:
```
[2020-03-03T18:20:45Z ERROR policy_engine::graph] Error serving request ' HttpRequest HTTP/1.1 GET:/api/upgrades_info/v1/graph   query: ?"channel=test&arch=amd64"   headers:     "accept": "application/json"     "x-forwarded-proto": "http"     "x-forwarded-for": "18.185.55.153"     "accept-encoding": "gzip"     "forwarded": "for=18.185.55.153;host=cincinnati-policy-engine-cincinnati-e2e.apps.vrutkovs.devcluster.openshift.com;proto=http;proto-version=\"\""     "x-vegeta-seq": "14846"     "user-agent": "Go-http-client/1.1"     "host": "cincinnati-policy-engine-cincinnati-e2e.apps.vrutkovs.devcluster.openshift.com"     "x-forwarded-host": "cincinnati-policy-engine-cincinnati-e2e.apps.vrutkovs.devcluster.openshift.com"     "x-forwarded-port": "80" ' from '10.128.2.12:56468': FailedUpstreamFetch("408 Request Timeout")
[2020-03-03T18:20:45Z ERROR policy_engine::graph] Error serving request ' HttpRequest HTTP/1.1 GET:/api/upgrades_info/v1/graph   query: ?"channel=b&arch=amd64"   headers:     "accept": "application/json"     "x-forwarded-proto": "http"     "x-forwarded-for": "18.185.55.153"     "accept-encoding": "gzip"     "forwarded": "for=18.185.55.153;host=cincinnati-policy-engine-cincinnati-e2e.apps.vrutkovs.devcluster.openshift.com;proto=http;proto-version=\"\""     "x-vegeta-seq": "14860"     "user-agent": "Go-http-client/1.1"     "host": "cincinnati-policy-engine-cincinnati-e2e.apps.vrutkovs.devcluster.openshift.com"     "x-forwarded-host": "cincinnati-policy-engine-cincinnati-e2e.apps.vrutkovs.devcluster.openshift.com"     "x-forwarded-port": "80" ' from '10.128.2.12:56476': FailedUpstreamFetch("408 Request Timeout")
[2020-03-03T18:20:45Z ERROR policy_engine::graph] Error serving request ' HttpRequest HTTP/1.1 GET:/api/upgrades_info/v1/graph   query: ?"channel=test&arch=amd64"   headers:     "accept": "application/json"     "x-forwarded-proto": "http"     "x-forwarded-for": "18.185.55.153"     "accept-encoding": "gzip"     "forwarded": "for=18.185.55.153;host=cincinnati-policy-engine-cincinnati-e2e.apps.vrutkovs.devcluster.openshift.com;proto=http;proto-version=\"\""     "x-vegeta-seq": "23999"     "user-agent": "Go-http-client/1.1"     "host": "cincinnati-policy-engine-cincinnati-e2e.apps.vrutkovs.devcluster.openshift.com"     "x-forwarded-host": "cincinnati-policy-engine-cincinnati-e2e.apps.vrutkovs.devcluster.openshift.com"     "x-forwarded-port": "80" ' from '10.128.2.12:54156': FailedUpstreamFetch("error sending request for url (http://localhost:8080/v1/graph): connection error: Connection reset by peer (os error 104)")
```

### CODE REVIEW

File: cincinnati/src/plugins/internal/cincinnati_graph_fetch.rs [hunk 1]
```diff
@@ -81,10 +84,15 @@ impl CincinnatiGraphFetchPlugin {
             registry.register(Box::new(http_upstream_errors_total.clone()))?;
         };
 
+        let client = reqwest::ClientBuilder::new()
+            .build()
+            .map_err(|e| GraphError::FailedUpstreamRequest(e.to_string()))?;
```
Comment (vrutkovs) [inline cincinnati/src/plugins/internal/cincinnati_graph_fetch.rs]: This is probably a wrong error to throw - any suggestions?

Comment (steveej) [reply cincinnati/src/plugins/internal/cincinnati_graph_fetch.rs]: This happens at initialization time before any plugin is executed. Other plugins don't set s a specific error here. How about a simple
```suggestion
            .context("Building reqwest client")?;
```

**Review by LalatenduMohanty — APPROVED** [review]
/hold for another review from Stefan or someone else
/lgtm

**Comment by jeremyeder** [comment]
Nice.

File: cincinnati/src/plugins/internal/cincinnati_graph_fetch.rs (hunk 1 above)
Comment (steveej) [reply cincinnati/src/plugins/internal/cincinnati_graph_fetch.rs]: Somehow my suggested change seems to be reverted again.

```suggestion
            .context("Building reqwest client")?;
```

**Review by steveej — CHANGES_REQUESTED** [review]

File: cincinnati/src/plugins/internal/cincinnati_graph_fetch.rs (hunk 1 above)
Comment (vrutkovs) [reply cincinnati/src/plugins/internal/cincinnati_graph_fetch.rs]: Sorry, must have overwritten that with a force push. Implemented that, thanks!

**Review by steveej — APPROVED** [review]
Thanks!

/lgtm

**Pull request merged**